core/src/main/java/org/apache/lucene/document/LatLonPointDistanceQueryFields.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.lucene.document;
+
+import org.apache.lucene.search.Query;
+
+/**
+ * Workaround to get access to {@link LatLonPointDistanceQuery} properties due to the fact that this class
```

Comment:
I think the reason why this work is because this class is loaded by parent class loader that also loads the Lucene jars and the module class loader just uses that. If this class would be in the class loader used the percolator module then this hack would fail (integ tests would tell us that). What java9 change will not allow this hack to work?

I will replace this hack once the LatLonPointDistanceQuery and LatLonPointInPolygonQuery classes have been made public.